packages/rsocket-core/src/RSocketResumableTransport.js

```diff
@@ -275,13 +275,18 @@ export default class RSocketResumableTransport implements DuplexConnection {
     if (this._isTerminated()) {
       return;
     }
-    if (error) {
-      this._setConnectionStatus({error, kind: 'ERROR'});
-    } else {
-      this._setConnectionStatus(CONNECTION_STATUS.CLOSED);
-    }
+
+    const status = error ? {error, kind: 'ERROR'} : CONNECTION_STATUS.CLOSED;
+    this._setConnectionStatus(status);
+
     const receivers = this._receivers;
-    receivers.forEach(r => r.onComplete());
+    receivers.forEach(subscriber => {
+      if (error) {
+        subscriber.onError(error);
```

Comment:
I'm wondering if we need to also call `subscriber.onComplete()` here since the connection has essentially failed. Otherwise, the client will need to know that if they receive an error during connection that they should cancel their own subscription, rather than the server doing it for them as it would have previously. Please let me know if that doesn't track.

Reply:
Sorry if I'm misunderstanding your concern (I'm a newcomer here, after all), but I think `onError` should already end the subscription as it's a terminal event. The reactive stream contract as I understand it is `onSubscribe onNext* (onError|onComplete)`. And, in fact, the subscriber implementations for `Flowable` and `Single` just log a warning and return if you try calling `onComplete` after `onError`.

As for closing the actual transport connection, that's what the call to `_disconnect()` below should be doing, I think?

Reply:
Once again, you are entirely correct. I mistook this class as being a server implementation, not a client, which caused some misunderstanding. I'm also new to the project, and learning :).

Reply:
Ah, that makes a lot of sense, haha! No worries! Thanks for taking the time to look at this! :)